lib/spack/spack/build_systems/python.py

```diff
@@ -284,76 +249,124 @@ def test(self):
+
+        This is usually the directory containing one of the following files:
+
+        * ``pyproject.toml``
+        * ``setup.cfg``
+        * ``setup.py``
         """
-        conflicts = list(dst for src, dst in merge_map.items() if os.path.exists(dst))
+        return self.pkg.stage.source_path
 
-        if conflicts and self.py_namespace:
-            ext_map = view.extensions_layout.extension_map(self.extendee_spec)
-            namespaces = set(x.package.py_namespace for x in ext_map.values())
-            namespace_re = r"site-packages/{0}/__init__.py".format(self.py_namespace)
-            find_namespace = match_predicate(namespace_re)
-            if self.py_namespace in namespaces:
-                conflicts = list(x for x in conflicts if not find_namespace(x))
+    def config_settings(self, pkg, spec, prefix):
+        """Configuration settings to be passed to the PEP 517 build backend.
+        Requires pip 22.1+, which requires Python 3.7+.
 
-        return conflicts
+        Args:
+            spec (spack.spec.Spec): build spec
+            prefix (spack.util.prefix.Prefix): installation prefix
 
-    def add_files_to_view(self, view, merge_map, skip_if_exists=True):
-        bin_dir = self.spec.prefix.bin
-        python_prefix = self.extendee_spec.prefix
-        python_is_external = self.extendee_spec.external
-        global_view = same_path(python_prefix, view.get_projection_for_spec(self.spec))
-        for src, dst in merge_map.items():
-            if os.path.exists(dst):
-                continue
-            elif global_view or not path_contains_subdirectory(src, bin_dir):
-                view.link(src, dst)
-            elif not os.path.islink(src):
-                shutil.copy2(src, dst)
-                is_script = is_nonsymlink_exe_with_shebang(src)
-                if is_script and not python_is_external:
-                    filter_file(
-                        python_prefix,
-                        os.path.abspath(view.get_projection_for_spec(self.spec)),
-                        dst,
-                    )
-            else:
-                orig_link_target = os.path.realpath(src)
-                new_link_target = os.path.abspath(merge_map[orig_link_target])
-                view.link(new_link_target, dst)
+        Returns:
+            dict: dictionary of KEY, VALUE settings
+        """
+        return {}
 
-    def remove_files_from_view(self, view, merge_map):
-        ignore_namespace = False
-        if self.py_namespace:
-            ext_map = view.extensions_layout.extension_map(self.extendee_spec)
-            remaining_namespaces = set(
-                spec.package.py_namespace for name, spec in ext_map.items() if name != self.name
-            )
-            if self.py_namespace in remaining_namespaces:
-                namespace_init = match_predicate(
-                    r"site-packages/{0}/__init__.py".format(self.py_namespace)
+    def install_options(self, pkg, spec, prefix):
```

Comment:
Why is `pkg` now an option? Also, needs to go into docstrings.

Reply:
Because the builder **is not** the package, but it **has a** package. So, I translated `self, spec, prefix` to `self, pkg, spec, prefix` (since `self` now has another meaning).

Reply:
Is `pkg == self.pkg`? If that's the case, I would remove this. Unlike `spec` and `prefix`, I don't think `pkg` is commonly used in packages. This relates back to our discussion on `self` vs. `self, spec, prefix`, although I think there's still room for common ground here.